### ISSUE TITLE
pmemobj: it is possible to assign one object handle to the other object handle of different type

### ISSUE THREAD

**Issue by michalsnochowski** [pmem/issues#65]
Steps to reproduce:
```
   struct test1 {
        int field1;
        int field2;
    };

    struct test2 {
        int field1;
    };

    OID_TYPE(struct test1) t1;
    OID_TYPE(struct test2) t2;

    OID_ASSIGN_TYPED(t1, t2);
```
Current result:
Above code compiles without any errors/warnings

Expected result:
As it is described in the manpage:
```
OID_ASSIGN_TYPED(OID_TYPE lhs, OID_TYPE rhs)
The OID_ASSIGN_TYPED macro assigns an object handle specified by rhs to the object handle
variable lhs with the static type checking. Both lhs and rhs must be the object handles of the same
type.  Otherwise a compile-time error is generated.
```

Reproduced on Red Hat Enterprise Linux Server release 7.1 (Maipo), kernel 3.10, gcc 4.8.3

**Comment by krzycz** [comment]
The problem reproduces only if compiler includes *libpmemobj.h* from a system header directory (like */usr/include*), which is a typical scenario for end users. By default, all the warnings are suppressed when processing system headers, so the compile-time type-checking in *OID_ASSIGN_TYPED* will not work unless the program is compiled with *-Wsystem-headers* flag.

**Comment by pbalcer** [comment]
You can close this one - we got rid of this macro.

**Comment by ldorau** [comment]
... so it is fixed by the commit 6be7be9437846ffd6afed236cae03c5d0f2c0b85:
https://github.com/pmem/nvml/commit/6be7be9437846ffd6afed236cae03c5d0f2c0b85

**Comment by michalsnochowski** [comment]
Verified on 0.1+b16-197-ga409fd2
